custom_components/tahoma/tahoma_device.py

```diff
@@ -109,10 +113,26 @@ def device_info(self) -> Dict[str, Any]:
         manufacturer = self.select_state(CORE_MANUFACTURER_NAME_STATE) or "Somfy"
         model = self.select_state(CORE_MODEL_STATE) or self.device.widget
 
+        name = self.name
+        device_url = self.device_url
+        if "#" in self.device_url:
+            device_url = self.device.deviceurl.split("#", 1)[0]
+            entity_registry = self.hass.data["entity_registry"]
+            name = next(
+                (
+                    entry.original_name
+                    for entity_id, entry in entity_registry.entities.items()
+                    if entry.unique_id == f"{device_url}#1"
```

Comment:
Where the entry.unique_id come from?

Reply:
note sure I understand the question?

Reply:
it comes from `tahoma_device`
```py
    @property
    def unique_id(self) -> str:
        """Return a unique ID."""
        return self.device.deviceurl
```

Reply:
It was to be sure. I was confused between entity_id and entry. Sorry.